core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -344,14 +344,15 @@ private void addTablePrivilegesRecords(QualifiedTablePrefix prefix)
 
     private void addRolesRecords()
     {
+        Optional<String> catalogName = metadata.isCatalogManagedSecurity(session, this.catalogName) ? Optional.of(this.catalogName) : Optional.empty();
```

Comment:
Might be cleaner to use `Optional.filter`

Reply:
I tried that and it was less awesome